src/cps/config.hpp.in

```diff
@@ -0,0 +1,5 @@
+#pragma once
+
+#define CPS_CONFIG_VERSION "${CMAKE_PROJECT_VERSION}"
```

Comment:
I'm not a huge fan of putting version information in header files as it pessimizes build caching in general, though keeping feature parity across CMake and meson is more important for this PR to land. A followon PR to consider another approach would be interesting to me. Eventually.

Reply:
Making an issue to follow up on this